### PR TITLE
feat: Support self-hosted/custom git providers

### LINKED ISSUE #137: Build error with custom/private/self-hosted git server
I'm currently playing around with `mdxts`.

While running the `build` command with the following `next.config.mjs`:

```js
// next.config.mjs

import { createMdxtsPlugin } from 'mdxts/next'

const withMdxts = createMdxtsPlugin({
  theme: 'github-dark',
  siteUrl: 'https://git.<company-host>/<namespace>/documentation',
  gitSource: 'https://git.<company-host>/<namespace>/documentation',
  gitBranch: 'main'
})
/** @type {import('next').NextConfig} */
const nextConfig = {
  eslint: {
    ignoreDuringBuilds: true
  },
  output: "export"
}

export default withMdxts(nextConfig)
```

I get the following error:

```bash
Failed to compile.

./docs/test-file.mdx
Error: test-file.mdx:Error: Git provider not recognized for https://git.<company-host>/<namespace>/documentation
```

Based on the documentation and the related source code:
* https://www.mdxts.dev/packages/components/git-provider
* https://github.com/souporserious/mdxts/blob/main/packages/mdxts/src/components/GitProvider.tsx

it seems that private git repositories are currently not supported.
In our case it's a self-hosted gitlab instance.

## Possible solution

* Add the relevant components to the `mdx-components.tsx` so we can override them
* Add a new prop ( .e.g. `gitProvider` ) to `createMdxtsPlugin` where we can configure the provider ( gitlab / github / bitbucket )
  * this configuration will be checked in the `GitProvider.tsx`
  * if not defined, try to fetch the provider from the url ( current behavior )

### PR DESCRIPTION
**Pull request by noxify**
Previously only `github.com`, `gitlab.com` and `bitbucket.org` were supported as git provider.
This merge request adds the functionality to set the git provider manually.

fixes #137

### CODE REVIEW

**Comment by noxify** [comment]
Updated the `getGitFileUrl` function and added also some unit tests.
I hope it's ok that I have updated the logic inside the function and using `path.join()` isn't a problem to build the url path.

Reason why I did this change: While creating the unit tests I got some errors with the expected vs. received url.
This was based on the `filePath`. 
Now it doesn't matter if you have a leading slash or not.

Also updated the input for the `switch case` block.

Let me know that you think about these changes @souporserious.

File: packages/mdxts/src/utils/get-git-file-url.ts [hunk 1]
```diff
@@ -1,35 +1,41 @@
...
+  gitProvider?: string
 ): string {
   const url = new URL(gitSource)
   let fileUrl: string
 
-  switch (url.hostname) {
-    case 'github.com':
-      fileUrl = `${gitSource}/blob/${gitBranch}/${filePath}`
+  const provider =
+    gitProvider || gitSource.replace(/^https?:\/\/(?:.*\.)?(.*?)\..*/gm, `$1`)
```
Comment (noxify) [inline packages/mdxts/src/utils/get-git-file-url.ts:16]: alternative would be using a package which resolves the domain name - but for our case, I think a "simple" regex should be enough

**Comment by noxify** [comment]
Based on the amount of arguments for `getGitFileUrl`, maybe it makes sense to use an object input.
This would introduce a breaking change, but makes it easier to configure.

Example
```ts
// old
getGitFileUrl({
        '/directory-name/file-name',
        undefined,
        undefined,
        'https://self-hosted.git.company.tld/owner-name/repo-name',
        undefined,
        'bitbucket'
})

// new
getGitFileUrl({
  filePath: '/directory-name/file-name',
  gitSource: 'https://self-hosted.git.company.tld/owner-name/repo-name',
  gitProvider: 'bitbucket'
})

```

**Comment by noxify** [comment]
Updated the relevant parts and added also a new example.

But while running 

```
pnpm build --filter=@examples/custom-git-provider
```

I get the same error - Added temporary some debug prints and it seems that the `gitProvider` value isn't passed down to the `getGitFileUrl`.

I assumed that `process.env.MDXTS_GIT_PROVIDER` will be used as fallback in the `getSourcePath` function, but it's currently `undefined`.

Any idea how to solve this ?

**Comment by noxify** [comment]
Ok, got it - added the `gitProvider` prop where it was required ( = where `getSourcePath` was called ) - Now the build is working

**Comment by noxify** [comment]
Ok - for me it looks good.

* clone my forked repo / [checkout the PR branch](https://docs.github.com/de/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/checking-out-pull-requests-locally)
* Run `pnpm build --filter=@examples/custom-git-provider`
* Run `pnpm -F custom-git-provider preview`
* Open the browser
* Click the link 
* You will see the "gitlab" icon and as link "https://git.company.tld/souporserious/mdxts"

File: packages/mdxts/src/utils/get-git-file-url.ts (hunk 1 above)
Comment (souporserious) [reply packages/mdxts/src/utils/get-git-file-url.ts:16]: Yeah, simple is great! It can always be adapted later if needed.

**Review by souporserious — APPROVED** [review]
This looks great 🎉 thank you for adding tests!

**Pull request merged**